### PR TITLE
[MOD-15932] trie: fix rangeIterate double-emit when boundary is proper prefix of child label

### PR DESCRIPTION
**Pull request by memark**
## Describe the changes in the pull request

1. **Current**: When `Trie_IterateRange` is called with a min/max value that is a proper prefix of a child node's collapsed radix label, `rangeIterate` (`src/trie/trie_node.c`) visits the boundary subtree twice. The dedicated `beginEqIdx`/`endEqIdx` recursion processes it once, then the immediately-following `for (ii = beginIdx; ii <= endIdx; ++ii)` loop processes it again — because `rsb_gt`/`rsb_lt` return the same index as the prefix-equal one (e.g. `"b" < "ban"` under exact compare, so `rsb_gt` includes `"ban"` even though we already recursed into it).
2. **Change**: After computing `beginIdx`/`endIdx`, trim them to skip past the prefix-boundary child the surrounding code already recurses on:
    ```
    if (beginEqIdx != -1 && beginIdx <= beginEqIdx) beginIdx = beginEqIdx + 1;
    if (endEqIdx   != -1 && endIdx   >= endEqIdx)   endIdx   = endEqIdx   - 1;
    ```
3. **Outcome**: Each terminal in range is yielded exactly once. Affects text-field lex range queries (`Query_EvalLexRangeNode`); tag lex ranges go through a different trie (`TrieMap_IterateRange`) and were never affected.

#### Which additional issues this PR fixes

None.

#### Main objects this PR modified

1. `rangeIterate` in `src/trie/trie_node.c`
2. New regression test `TrieTest::testRangeBoundaryPrefix` in `tests/cpptests/test_cpp_trie.cpp` — exercises both min-only (`[b, +inf)`) and max-only (`(-inf, banb]`) shapes against a fixture with a multi-char collapsed label (`apple, banana, band, bandana, cherry, date`). Uses a fresh `DupTrackingCtx` callback (counts duplicates via `set::insert(...).second`) since the existing `rangeFunc`'s `assert()` compiles out under NDEBUG.

#### Mark if applicable

- [ ] This PR introduces API changes
- [ ] This PR introduces serialization changes

#### Release Notes

- [ ] This PR requires release notes
- [x] This PR does not require release notes

---

> [!NOTE]
> **Medium Risk**
> Correctness fix for lex range queries on the term trie; behavior changes for edge cases that previously duplicated results, with targeted regression tests and no API changes.
> 
> **Overview**
> Fixes **duplicate terms** in lexicographic trie range scans when the min or max bound is a **proper prefix** of a child’s collapsed radix label (e.g. min `b` vs child label `ban`).
> 
> In `rangeIterate` (`trie_node.c`), after `rsb_gt` / `rsb_lt` pick the bulk child span, the loop now **skips** children already handled by the `beginEqIdx` / `endEqIdx` prefix recursions (`beginIdx = beginEqIdx + 1`, `endIdx = endEqIdx - 1`), so each subtree is walked once.
> 
> Adds **`TrieTest::testRangeBoundaryPrefix`** with textual terms and min-only / max-only ranges; duplicate hits would trip `rangeFunc`’s insert assert. Affects **text-field lex range** evaluation (`Trie_IterateRange` / `Query_EvalLexRangeNode`); tag lex ranges use a different path.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 9efa7120da34a33208ac5a936afba6dfe95d4102. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Review by ThierryBerger — DISMISSED** [review]
Tests are sound, the change makes sense.

Approved, but may be worth double checking the sonarqube comments, which are mostly cpp idiomaticness feedbacks.

File: tests/cpptests/test_cpp_trie.cpp [hunk 1]
```diff
@@ -162,6 +162,75 @@ TEST_F(TrieTest, testBasicRangeWithScore) {
...
+
+static DupTrackingCtx trieIterRangeTracked(Trie *t, const char *begin, const char *end) {
+  rune r1[256] = {0}, r2[256] = {0};
+  size_t nr1 = 0, nr2 = 0;
+  rune *r1Ptr = nullptr, *r2Ptr = nullptr;
+  int n1 = -1, n2 = -1;
+  if (begin) {
+    nr1 = strToRunesN(begin, strlen(begin), r1);
+    r1Ptr = r1;
+    n1 = nr1;
```
Comment (raz-mon) [inline tests/cpptests/test_cpp_trie.cpp]: Why do we need `n1, n2`?

File: tests/cpptests/test_cpp_trie.cpp [hunk 2]
```diff
@@ -162,6 +162,75 @@ TEST_F(TrieTest, testBasicRangeWithScore) {
...
+
+static int dupTrackingFunc(const rune *u16, size_t nrune, void *ctx, void *payload, size_t numDocsInTerm) {
+  size_t n;
+  char *s = runesToStr(u16, nrune, &n);
+  std::string xs(s, n);
+  free(s);
+  DupTrackingCtx *dc = (DupTrackingCtx *)ctx;
+  if (!dc->seen.insert(xs).second) {
+    dc->duplicates++;
+  }
```
Comment (raz-mon) [inline tests/cpptests/test_cpp_trie.cpp]: This should be already caught in:
```
static int rangeFunc(const rune *u16, size_t nrune, void *ctx, void *payload, size_t numDocsInTerm) {
  size_t n;
  char *s = runesToStr(u16, nrune, &n);
  std::string xs(s, n);
  free(s);
  ElemSet *e = (ElemSet *)ctx;
  assert(e->end() == e->find(xs));
  e->insert(xs);
  return REDISEARCH_OK;
}
```
Will the addition of the new test case suffice? Let's make sure it fails there.

File: tests/cpptests/test_cpp_trie.cpp (hunk 1 above)
Comment (memark) [reply tests/cpptests/test_cpp_trie.cpp]: We don't. Removed it.

File: tests/cpptests/test_cpp_trie.cpp (hunk 2 above)
Comment (memark) [reply tests/cpptests/test_cpp_trie.cpp]: It will. Simplified it a bit.

**Review by raz-mon — APPROVED** [review]
